tools/biosnoop.bt

```diff
@@ -12,32 +13,36 @@
 
 BEGIN
 {
-	printf("%-12s %-16s %-6s %7s\n", "TIME(ms)", "COMM", "PID", "LAT(ms)");
+	printf("%-12s %-7s %-16s %-6s %7s\n", "TIME(ms)", "DISK", "COMM", "PID", "LAT(ms)");
 }
 
 kprobe:blk_account_io_start
 {
 	@start[arg0] = nsecs;
 	@iopid[arg0] = pid;
 	@iocomm[arg0] = comm;
+	@disk[arg0] = ((struct request *)arg0)->rq_disk->disk_name;
 }
 
 kprobe:blk_account_io_done
 /@start[arg0] != 0 && @iopid[arg0] != 0 && @iocomm[arg0] != ""/
+
 {
 	$now = nsecs;
-	printf("%-12u %-16s %-6d %7d\n",
-	    elapsed / 1e6, @iocomm[arg0], @iopid[arg0],
-	    ($now - @start[arg0]) / 1e6);
+	printf("%-12u %-7s %-16s %-6d %7d\n",
+	    elapsed / 1000000, @disk[arg0], @iocomm[arg0], @iopid[arg0],
+	    ($now - @start[arg0]) / 1000000);
```

Comment:
What's the reason for the `1e6` -> `1000000` changes?

Reply:
my compiler refused to parse 1e6 -- i'll revert this change if you think i should

Reply:
You're probably on an older bpftrace. Scientific notation was merged in #1476.

Upstream tools should be targeting upstream bpftrace - we don't guarantee upstream tools are compatible across old versions of bpftrace.